src/main.rs

```diff
@@ -1,14 +1,46 @@
-use std::path::PathBuf;
+use std::{io::Write, path::PathBuf};
+
+use clap::Parser;
 
 use jakt::{Compiler, JaktError, Span};
 
+#[derive(Parser, Debug)]
+struct CLIArgs {
+    /// Binary directory
+    #[clap(long = "binary-directory", short = 'o')]
+    binary_directory: Option<PathBuf>,
+
+    /// Files to compile
+    #[clap(min_values(1), required = true)]
+    files: Vec<PathBuf>,
+}
+
 fn main() -> Result<(), JaktError> {
+    let cli = CLIArgs::parse();
+
+    let binary_directory = match cli.binary_directory {
+        Some(dir) => dir,
+        None => {
+            let mut dir = std::env::current_dir().expect("Current directory unusable!");
+            dir.push("build");
+            dir
```

Comment:
Can you not collapse this into one statement, or does that create a dropped temporary variable?

(to be fair I haven't worked much with Rust's path API yet)

Reply:
push doesn't return a value, if there's a better way i'm all ears

Reply:
Sadly those APIs don't return this and the struct itself is the right choice. Then I'm fine with it.